doc/delivery-monitoring-protocol.md

```diff
@@ -0,0 +1,51 @@
+
+## PROTOCOL DESCRIPTION
+### SPECIAL FRAGMENTS
+- `ACK`: does not contain any information, has to be sent in response to [`ACK`](#special-fragments) from the other side.
+- `SYN`: contains [`PTR`](#data-structures) and [`CNT`](#data-structures), and has to be sent primarily after [`TRANSPORT`](#terms) restoration from both sides.
+  
+### DATA STRUCTURES
+- `S[0]`, `S[1]`: two "storages" of sent [`FRAGMENT`](#terms)s. (The "storage" is only conditional. The main thing is that you should have access to the already sent [`FRAGMENT`](#terms)s, divided into two parts as the protocol requires.)
+- `PTR`: a boolean pointer to which "storage" the [`FRAGMENT`](#terms)s should be written.
+- `FRS`: a boolean field, `1` for the one who sends an [`ACK`](#special-fragments) first.
+- `CNT`: an unsigned integer counter of the received data [`FRAGMENT`](#terms)s.
+
+### PROTOCOL STAGES
+#### PREPARATION
+  Before the start of the transferring [`FRAGMENT`](#terms)s both sides have to initialize the necessary data structures with default values (empty storages([`S[0]`](#data-structures), [`S[1]`](#data-structures)), [`PTR`](#data-structures)`= 0`, [`CNT`](#data-structures)`= 0`), decide which side sends an [`ACK`](#special-fragments) first and depending on this, initialize the value of the [`FRS`](#data-structures): `1` if the side sends the [`ACK`](#special-fragments) first, `0` otherwise.
+  
+  After it you can start sending [`FRAGMENT`](#terms)s with writing each [`FRAGMENT`](#terms) to the [`S[PTR]`](#data-structures) on the sender's side, and send the first [`ACK`](#special-fragments).
+
+#### NORMAL MODE
+  When side sends [`FRAGMENT`](#terms), it has to write a [`FRAGMENT`](#terms) to the [`S[PTR]`](#data-structures). When receiver side gets [`FRAGMENT`](#terms), it has to increment [`CNT`](#data-structures). When anyone gets [`ACK`](#special-fragments) it has to set [`CNT`](#data-structures)`= 0`, clear [`S[!PTR]`](#data-structures) and then send the [`ACK`](#special-fragments) back and at the same time negate the [`PTR`](#data-structures).
+
+| N step | Transition | N + 1 step |
+|:--------------:|:-------------:|:----------------:|
+| [`PTR`](#data-structures), [`CNT`](#data-structures) | send [`FRAGMENT`](#terms) | [`PTR`](#data-structures), [`CNT`](#data-structures) |
+| [`PTR`](#data-structures), [`CNT`](#data-structures) | recieved [`FRAGMENT`](#terms) | [`PTR`](#data-structures), [`CNT`](#data-structures) + 1 |
+| [`PTR`](#data-structures), [`CNT`](#data-structures) | send [`ACK`](#special-fragments) | [`!PTR`](#data-structures), [`CNT`](#data-structures) |
+| [`PTR`](#data-structures), [`CNT`](#data-structures) | recieved [`ACK`](#special-fragments) | [`PTR`](#data-structures), `0` |
+
+![dmp_normal_mode_image](https://user-images.githubusercontent.com/31672093/57922020-7471d700-78a7-11e9-8ea8-a86cb0c6485b.gif)
+
+#### RESTORE MODE
```

Comment:
Is it possible for `Transport` to die before any `ACK` will be sent?
F1-> 
              <-F1
---TRANSPORT DIED--

Reply:
The rule is: if you ACK, you must send ACK (even if the transport has already fallen). In the case of the first posting the same, you must send ACK and even if the transport has fallen.

Reply:
So do I understand correctly that you must have unresponded ACK to start restoration?

Reply:
ACK is sent from both sides. The fact of the fall of the connection can state both sides. I would not say that there is someone who initiated the recovery.

Reply:
I mean such test
javascript```
{
  const a = new Restore(true);
  const b = new Restore(false);
  b.recieveData(a.sendData('c1'));
  a.recieveData(b.sendData('r1'));

  console.log(a.restore(b));
  console.log(b.restore(a));
}

```